tiledb/sm/query/reader.cc

```diff
@@ -1569,7 +1591,7 @@ Status Reader::dedup_result_coords(
   while (it != coords_end) {
     auto next_it = skip_invalid_elements(std::next(it), coords_end);
     if (next_it != coords_end && it->same_coords(*next_it)) {
-      if (it->tile_->frag_idx() < next_it->tile_->frag_idx()) {
+      if (get_timestamp(*it) < get_timestamp(*next_it)) {
```

Comment:
Just making sure: this means that results with same `coords` are ordered by the time they were written and no two fragments can be written at the same time?

Reply:
They are not sorted by timestamps. 'it' always stays with the value that has the highest timestamp. If two fragments are written at the same time, one of the coordinates will be chosen, but it is not defined anywhere in our storage format which one.